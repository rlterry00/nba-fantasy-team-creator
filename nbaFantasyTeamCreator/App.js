import React, {useState, useEffect} from 'react';
import {
  SafeAreaView,
  ScrollView,
  StyleSheet,
  Text,
  View,
  FlatList,
  TouchableOpacity,
  Button,
  Modal,
  Image,
  TextInput,
  Alert,
} from 'react-native';

const App = () => {
  const [playersList, setPlayersList] = useState([]);
  const [filteredPlayersList, setFilteredPlayersList] = useState([]);
  const [playersVisible, setPlayersVisible] = useState(false);
  const [teamSetUp, setTeamSetUp] = useState(false);
  const [useDefaultImage, setUseDefaultImage] = useState(true);
  const [defaultImage, setDefaultImage] = useState('./assets/avatar.png');
  const [teamName, setTeamName] = useState('');
  const [cityName, setCityName] = useState('');
  const [teamRoster, setTeamRoster] = useState([]);
  useEffect(() => {
    const getPlayersList = async () => {
      return fetch('https://data.nba.net/10s/prod/v1/2021/players.json')
        .then(response => response.json())
        .then(json => {
          const data = json.league.sacramento;
          // console.log(data);
          setPlayersList(data);
        })
        .catch(error => {
          console.error(error);
        });
    };
    getPlayersList();
  }, []);

  const addPlayers = (
    personId,
    playerImage,
    playerPosition,
    firstName,
    lastName,
  ) => {
    if (teamRoster.filter(x => x.personId === personId).length > 0) {
      Alert.alert('Player already chosen');
    } else if (
      teamRoster.filter(x => x.playerPosition === playerPosition).length > 0
    ) {
      Alert.alert(
        'You already have player in that position on your team. Remove previous player first.',
      );
    } else {
      return teamRoster.push({
        personId: personId,
        playerImage: playerImage,
        playerPosition: playerPosition,
        firstName: firstName,
        lastName: lastName,
      });
    }
  };

  const removePlayer = personId => {
    var rosterList = teamRoster.filter(x => {
      return x.personId !== personId;
    });
    console.log(rosterList);
    setTeamRoster(rosterList);
  };

  const filterPlayers = playerPosition => {
    var playerList = playersList.filter(x => {
      return x.teamSitesOnly.posFull === playerPosition;
    });
    setFilteredPlayersList(playerList);
  };

  const ListItem = ({item}) => {
    const personId = item.personId;
    const playerImage =
      'https://ak-static.cms.nba.com/wp-content/uploads/headshots/nba/latest/260x190/' +
      personId +
      '.png';
    const playerPosition = item.teamSitesOnly.posFull;
    const firstName = item.firstName;
    const lastName = item.lastName;
    return (
      <View style={styles.listContainer}>
        <View>
          <Image
            style={{width: 100, height: 100, marginLeft: 5}}
            defaultSource={require('./assets/avatar.png')}
            source={{uri: playerImage}}
            resizeMode="cover"
          />
          <View style={styles.nameRow}>
            <Text style={styles.nameTitle}>{firstName}</Text>
            <Text style={styles.nameTitle}>{lastName}</Text>
          </View>
          <Text style={styles.listItem}>Position: {playerPosition}</Text>
          <Text style={styles.listItem}>
            Date of Birth: {item.dateOfBirthUTC}
          </Text>
          <Text style={styles.listItem}>
            Height: {item.heightFeet} Feet {item.heightInches} Inches
          </Text>
          <Text style={styles.listItem}>Weight: {item.weightPounds} lb</Text>
          <Text style={styles.listItem}>College: {item.collegeName}</Text>
          <Text style={styles.listItem}>
            NBA Debut Year: {item.nbaDebutYear}
          </Text>
          <Text style={styles.listItem}>
            Draft Pick Number: {item.draft.pickNum}
          </Text>
          <Text style={styles.listItem}>
            Draft Round Number: {item.draft.roundNum}
          </Text>
          <Text style={styles.listItem}>
            Draft Season Year: {item.draft.seasonYear}
          </Text>
          <View style={styles.button}>
            <Button
              onPress={() => {
                addPlayers(
                  personId,
                  playerImage,
                  playerPosition,
                  firstName,
                  lastName,
                ),
                  setPlayersVisible(false);
              }}
              title="Add Player"
              color="black"
            />
          </View>
        </View>
      </View>
    );
  };

  const rosterList = ({item}) => {
    const personId = item.personId;
    const playerImage = item.playerImage;
    const playerPosition = item.playerPosition;
    const firstName = item.firstName;
    const lastName = item.lastName;
    return (
      <View style={styles.listContainer}>
        <View>
          <View style={styles.titleViewRowRoster}>
            <Text style={styles.sectionTitle}>{cityName}</Text>
            <Text style={styles.sectionTitle}>{teamName}</Text>
          </View>
          <Text style={styles.sectionTitle}>Position: {playerPosition}</Text>
          <View style={styles.nameRow}>
            <Image
              style={{width: 100, height: 100, marginLeft: 5}}
              defaultSource={require('./assets/avatar.png')}
              source={
                useDefaultImage
                  ? require('./assets/avatar.png')
                  : {uri: playerImage}
              }
              resizeMode="cover"
            />

            <Text style={styles.sectionTitle}>{firstName}</Text>
            <Text style={styles.sectionTitle}>{lastName}</Text>
          </View>

          <View style={styles.button}>
            <Button
              onPress={() => removePlayer(personId)}
              title="Remove Player"
              color="black"
            />
          </View>
        </View>
      </View>
    );
  };

  return (
    <SafeAreaView style={styles.container}>
      <View>
        <View style={styles.titleView}>
          <Text style={styles.title}>NBA Fantasy Team Creator</Text>
        </View>
      </View>
      <View>
        <View style={styles.titleView}>
          <Text style={styles.sectionTitle}>Create Team:</Text>
          <View style={styles.button}>
            <Button
              onPress={() => setTeamSetUp(true)}
              title="Add Team Name and City"
              color="black"
            />
          </View>
          <View style={styles.button}>
            <Button
              onPress={() =>
                teamName && cityName
                  ? setPlayersVisible(true)
                  : Alert.alert('Please choose team city and name first')
              }
              title="Select Players"
              color="black"
            />
          </View>
          <View style={styles.titleViewRow}>
            <Text style={styles.sectionTitle}>{cityName}</Text>
            <Text style={styles.sectionTitle}>{teamName}</Text>
          </View>
          <Text style={styles.sectionTitle}>Roster:</Text>
        </View>
        <FlatList
          data={teamRoster}
          renderItem={rosterList}
          keyExtractor={item => item.personId}
          horizontal={true}
        />
      </View>
      <View>
        <Modal
          animationType="slide"
          transparent={false}
          visible={playersVisible}
          onRequestClose={() => {
            setPlayersVisible(!playersVisible);
          }}>
          <View style={styles.container}>
            <View style={styles.titleView}>
              <Text style={styles.sectionTitle}>Player List</Text>
            </View>
            <TouchableOpacity
              style={styles.titleView}
              onPress={() => setPlayersVisible(!playersVisible)}>
              <Text style={styles.exitText}>Exit</Text>
            </TouchableOpacity>
            <View style={styles.titleView}>
              <Text style={styles.sectionTitle}>Filter By Position:</Text>
            </View>
            <View style={styles.titleViewRow}>
              <TouchableOpacity
                style={styles.titleView}
                onPress={() => {
                  var position = 'Forward';
                  filterPlayers(position);
                }}>
                <Text style={styles.positionText}>Forward</Text>
              </TouchableOpacity>
              <TouchableOpacity
                style={styles.titleView}
                onPress={() => {
                  var position = 'Forward-Center';
                  filterPlayers(position);
                }}>
                <Text style={styles.positionText}>Forward-Center</Text>
              </TouchableOpacity>
              <TouchableOpacity
                style={styles.titleView}
                onPress={() => {
                  var position = 'Center';
                  filterPlayers(position);
                }}>
                <Text style={styles.positionText}>Center</Text>
              </TouchableOpacity>
              <TouchableOpacity
                style={styles.titleView}
                onPress={() => {
                  var position = 'Guard';
                  filterPlayers(position);
                }}>
                <Text style={styles.positionText}>Guard</Text>
              </TouchableOpacity>
            </View>
            <FlatList
              data={
                filteredPlayersList.length > 0
                  ? filteredPlayersList
                  : playersList
              }
              renderItem={ListItem}
              keyExtractor={item => item.personId}
              horizontal={false}
            />
          </View>
        </Modal>
        <Modal
          animationType="slide"
          transparent={false}
          visible={teamSetUp}
          onRequestClose={() => {
            setTeamSetUp(!teamSetUp);
          }}>
          <View style={styles.container}>
            <View style={styles.titleView}>
              <Text style={styles.sectionTitle}>Team Set Up</Text>
            </View>
            <TouchableOpacity
              style={styles.titleView}
              onPress={() => setTeamSetUp(!teamSetUp)}>
              <Text style={styles.exitText}>Exit</Text>
            </TouchableOpacity>
            <Text style={styles.sectionTitle}>Team City: {cityName}</Text>
            <Text style={styles.sectionTitle}>Team Name: {teamName}</Text>
            <TextInput
              placeholder="Type in city name"
              autoCapitalize="words"
              onChangeText={text => setCityName(text)}
              style={styles.inputStyle}
            />
            <TextInput
              placeholder="Type in team name"
              autoCapitalize="words"
              onChangeText={text => setTeamName(text)}
              style={styles.inputStyle}
            />

            <View style={styles.button}>
              <Button
                onPress={() => setTeamSetUp(!teamSetUp)}
                title="Submit"
                color="black"
              />
            </View>
          </View>
        </Modal>
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: 60,
    marginLeft: 5,
    marginRight: 5,
  },
  listContainer: {
    flex: 1,
    margin: 10,
  },
  titleView: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  titleViewRow: {
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'row',
  },
  titleViewRowRoster: {
    flexDirection: 'row',
  },
  title: {
    fontSize: 30,
  },
  sectionTitle: {
    fontSize: 25,
    marginRight: 5,
    marginLeft: 5,
    marginTop: 10,
  },
  nameRow: {
    flexDirection: 'row',
    justifyContent: 'space-evenly',
    borderBottomWidth: 1,
    marginBottom: 5,
  },
  nameTitle: {
    fontSize: 20,
  },
  listItem: {
    marginLeft: 10,
  },
  button: {
    backgroundColor: 'gainsboro',
    borderRadius: 15,
    margin: 10,
  },
  exitText: {
    color: 'blue',
    fontSize: 15,
  },
  positionText: {
    color: 'blue',
    fontSize: 15,
    marginRight: 5,
  },
  inputStyle: {
    borderBottomWidth: 1,
    margin: 10,
    height: 40,
  },
});
export default App;
