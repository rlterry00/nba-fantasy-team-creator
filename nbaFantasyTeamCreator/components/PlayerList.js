import React, {useMemo} from 'react';
import {StyleSheet, Text, View, Button, Image} from 'react-native';

const PlayerList = props => {
  const personId = props.personId;
  const playerImage =
    'https://ak-static.cms.nba.com/wp-content/uploads/headshots/nba/latest/260x190/' +
    personId +
    '.png';
  return (
    <View style={styles.listContainer}>
      <View>
        <Image
          style={{width: 100, height: 100, marginLeft: 5}}
          defaultSource={require('../assets/avatar.png')}
          source={{uri: playerImage}}
          resizeMode="cover"
        />
        <View style={styles.nameRow}>
          <Text style={styles.nameTitle} testID="firstName">
            {props.firstName}
          </Text>
          <Text style={styles.nameTitle}>{props.lastName}</Text>
        </View>
        <Text style={styles.listItem}>Position: {props.playerPosition}</Text>
        <Text style={styles.listItem}>Date of Birth: {props.dateOfBirth}</Text>
        <Text style={styles.listItem}>
          Height: {props.heightFeet} Feet {props.heightInches} Inches
        </Text>
        <Text style={styles.listItem}>Weight: {props.weightPounds} lb</Text>
        <Text style={styles.listItem}>College: {props.collegeName}</Text>
        <Text style={styles.listItem}>
          NBA Debut Year: {props.nbaDebutYear}
        </Text>
        <Text style={styles.listItem}>
          Draft Pick Number: {props.draftPickNum}
        </Text>
        <Text style={styles.listItem}>
          Draft Round Number: {props.draftRoundNum}
        </Text>
        <Text style={styles.listItem}>
          Draft Season Year: {props.draftSeasonYear}
        </Text>
        <View style={styles.button}>
          <Button
            onPress={() => {
              props.onPress();
            }}
            title="Add Player"
            color="black"
          />
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  listContainer: {
    flex: 1,
    margin: 10,
  },
  nameRow: {
    flexDirection: 'row',
    justifyContent: 'space-evenly',
    borderBottomWidth: 1,
    marginBottom: 5,
  },
  nameTitle: {
    fontSize: 20,
  },
  listItem: {
    marginLeft: 10,
  },
  button: {
    backgroundColor: 'gainsboro',
    borderRadius: 15,
    margin: 10,
  },
});
export default PlayerList;
