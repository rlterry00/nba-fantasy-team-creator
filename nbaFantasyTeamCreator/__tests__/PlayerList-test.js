import 'react-native';
import React from 'react';
import Enzyme, {mount, shallow} from 'enzyme';
import PlayerList from '../components/PlayerList';
import App from '../App';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

Enzyme.configure({adapter: new Adapter()});

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  renderer.create(<PlayerList />);
});

test('matches snapshot', () => {
  const playerList = renderer.create(<PlayerList />).toJSON();
  expect(playerList).toMatchSnapshot();
});

test('firstName Text component exists', () => {
  const wrapper = shallow(<PlayerList />);
  expect(
    wrapper.findWhere(node => node.prop('testID') === 'firstName'),
  ).toExist();
});

test('firstName prop exist', () => {
  const wrapper = shallow(<App />);
  expect(wrapper.findWhere(node => node.prop('firstName'))).toExist();
});
