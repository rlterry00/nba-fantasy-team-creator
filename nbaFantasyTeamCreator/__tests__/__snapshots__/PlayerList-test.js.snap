// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "margin": 10,
    }
  }
>
  <View>
    <Image
      defaultSource={
        Object {
          "testUri": "../../../assets/avatar.png",
        }
      }
      resizeMode="cover"
      source={
        Object {
          "uri": "https://ak-static.cms.nba.com/wp-content/uploads/headshots/nba/latest/260x190/undefined.png",
        }
      }
      style={
        Object {
          "height": 100,
          "marginLeft": 5,
          "width": 100,
        }
      }
    />
    <View
      style={
        Object {
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-evenly",
          "marginBottom": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 20,
          }
        }
      />
      <Text
        style={
          Object {
            "fontSize": 20,
          }
        }
      />
    </View>
    <Text
      style={
        Object {
          "marginLeft": 10,
        }
      }
    >
      Position: 
    </Text>
    <Text
      style={
        Object {
          "marginLeft": 10,
        }
      }
    >
      Date of Birth: 
    </Text>
    <Text
      style={
        Object {
          "marginLeft": 10,
        }
      }
    >
      Height: 
       Feet 
       Inches
    </Text>
    <Text
      style={
        Object {
          "marginLeft": 10,
        }
      }
    >
      Weight: 
       lb
    </Text>
    <Text
      style={
        Object {
          "marginLeft": 10,
        }
      }
    >
      College: 
    </Text>
    <Text
      style={
        Object {
          "marginLeft": 10,
        }
      }
    >
      NBA Debut Year: 
    </Text>
    <Text
      style={
        Object {
          "marginLeft": 10,
        }
      }
    >
      Draft Pick Number: 
    </Text>
    <Text
      style={
        Object {
          "marginLeft": 10,
        }
      }
    >
      Draft Round Number: 
    </Text>
    <Text
      style={
        Object {
          "marginLeft": 10,
        }
      }
    >
      Draft Season Year: 
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "gainsboro",
          "borderRadius": 15,
          "margin": 10,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "black",
                },
              ]
            }
          >
            Add Player
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
